students/cfarris/lessons/lesson07/assignment/linear.py

```diff
@@ -0,0 +1,226 @@
+Imports 3 files, directs them to different collections
+Retrieves information necessary for HP Norton collection
+
+Room for improvement:
+This file could use more error checking of the imported data.
+pylint returns a rating of 9.56 after disabling three areas(see below)
+These improvements will be tabled for future edits as they aren't essential
+to the lesson 7 assignment requirements.
+"""
+import time
+import csv
+from pymongo import MongoClient
+
+# pylint: disable-msg=redefined-outer-name
+# pylint: disable-msg=invalid-name
+# pylint: disable-msg=too-many-locals
+
+
+class MongoDBConnection():
+    """Connect to MongoDB"""
+
+    def __init__(self, host='127.0.0.1', port=27017):
+
+        self.host = host
+        self.port = port
+        self.connection = None
+
+    def __enter__(self):
+        self.connection = MongoClient(self.host, self.port)
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.connection.close()
+
+
+def print_mdb_collection(db, collection_name):
+    """Prints the collection specified"""
+
+    for doc in collection_name.find():
+        print(doc)
+
+
+def import_data(db, customers=None, products=None, rentals=None, start_time=0):
+    """
+    Adds files to collection.
+    if the file/date type (k) fits into three categories
+    it will be added to that collection.
+    I created a thread for rentals as well,
+    but commented it out to adhere to instructions.
+    """
+
+    rentals_db = db['rentals_db']
+    customers_db = db['customers_db']
+    products_db = db['products_db']
+
+    data_list = []
+
+    # if statements were added to facilitate testing.
+    if products:
+        _read_csv(db, products_db, products, data_list, start_time)
+
+    if customers:
+        _read_csv(db, customers_db, customers, data_list, start_time)
+
+    if rentals:
+        _read_csv(db, rentals_db, rentals, data_list, start_time)
+
+    print("data_list: ", data_list)
+    return data_list
+
+
+def _read_csv(db, collection, csv_file, data_list=None, start_time=0):
+    '''
+    Opens csv files, creates a list of
+    dictionarys with matching keys from header to values
+    returns the dictionary and saves it to
+    different mongodb collections based upon the
+    file name given to main.
+    '''
+
+    if data_list is None:
+        data_list = []
+
+    with open(csv_file, 'r', encoding='utf-8-sig') as csv_file:
+        csv_reader = csv.reader(csv_file, delimiter=',')
+        header = True
+        first_line = []
+        num_processed = 0
+        coll_of_dicts = []
+        count_before_running = collection.count()
+        for row in csv_reader:
+            if header:
+                header = False
+                first_line = [value.strip() for value in row]
+            else:
+                num_processed += 1
+                new_dict = {}
+                str_row = []
+                str_row = [value.strip() for value in row]
```

Comment:
You don't have to initialize variables with an empty list, if you are doing a comprehension. The comprehension is all you need!